pkg/operator/cluster/ceph/mon/health.go

```diff
@@ -66,258 +59,6 @@ func (hc *HealthChecker) Check(stopCh chan struct{}) {
-	if err = c.startPods(mConf); err != nil {
-		return fmt.Errorf("failed to start new mon %s. %+v", m.Name, err)
-	}
-
-	// Only increment the max mon id if the new pod started successfully
-	c.maxMonID++
-
-	return c.removeMon(name)
-}
-
-func (c *Cluster) removeMon(name string) error {
-	logger.Infof("ensuring removal of unhealthy monitor %s", name)
-
-	// Remove the mon pod if it is still there
-	var gracePeriod int64
-	propagation := metav1.DeletePropagationForeground
-	options := &metav1.DeleteOptions{GracePeriodSeconds: &gracePeriod, PropagationPolicy: &propagation}
-	if err := c.context.Clientset.Extensions().ReplicaSets(c.Namespace).Delete(name, options); err != nil {
-		if errors.IsNotFound(err) {
-			logger.Infof("dead mon %s was already gone", name)
-		} else {
-			return fmt.Errorf("failed to remove dead mon pod %s. %+v", name, err)
-		}
-	}
-
-	// Remove the bad monitor from quorum
-	if err := removeMonitorFromQuorum(c.context, c.clusterInfo.Name, name); err != nil {
-		return fmt.Errorf("failed to remove mon %s from quorum. %+v", name, err)
-	}
-	delete(c.clusterInfo.Monitors, name)
-	// check if a mapping exists for the mon
-	if _, ok := c.mapping.Node[name]; ok {
-		nodeName := c.mapping.Node[name].Name
-		delete(c.mapping.Node, name)
-		// if node->port "mapping" has been created, decrease or delete it
-		if port, ok := c.mapping.Port[nodeName]; ok {
-			if port == mon.DefaultPort {
-				delete(c.mapping.Port, nodeName)
-			}
-			// don't clean up if a node port is higher than the default port, other
-			//mons could be on the same node with > DefaultPort ports, decreasing could
-			//cause port collsions
-			// This can be solved by using a map[nodeName][]int32 for the ports to
-			//even better check which ports are open for the HostNetwork mode
-		}
-	}
-
-	// Remove the service endpoint
-	if err := c.context.Clientset.CoreV1().Services(c.Namespace).Delete(name, options); err != nil {
-		if errors.IsNotFound(err) {
-			logger.Infof("dead mon service %s was already gone", name)
-		} else {
-			return fmt.Errorf("failed to remove dead mon pod %s. %+v", name, err)
-		}
-	}
-
-	if err := c.saveMonConfig(); err != nil {
-		return fmt.Errorf("failed to save mon config after failing over mon %s. %+v", name, err)
-	}
-
-	// make sure to rewrite the config so NO new connections are made to the removed mon
-	if err := WriteConnectionConfig(c.context, c.clusterInfo); err != nil {
-		return fmt.Errorf("failed to write connection config after failing over mon %s. %+v", name, err)
-	}
-
-	return nil
-}
-
-func removeMonitorFromQuorum(context *clusterd.Context, clusterName, name string) error {
-	logger.Debugf("removing monitor %s", name)
-	args := []string{"mon", "remove", name}
-	if _, err := client.ExecuteCephCommand(context, clusterName, args); err != nil {
-		return fmt.Errorf("mon %s remove failed: %+v", name, err)
-	}
-
-	logger.Infof("removed monitor %s", name)
+	// TODO Copy old/Reimplement health check logic for StatefulSet/Pods
```

Comment:
What does it mean to perform the health check now? Kubernetes failover of the mons would cover many scenarios. Would be good to list out the scenarios where statefulsets do not keep things healthy